packages/console/src/pages/ApiResourceDetails/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { RequestError } from '@/swr';
+
+import * as styles from './index.module.scss';
+
+type FormData = {
+  name: string;
+  accessTokenTtl: number;
+};
+
+const ApiResourceDetails = () => {
+  const location = useLocation();
+  const { id } = useParams();
+  const { data, error, mutate } = useSWR<Resource, RequestError>(id && `/api/resources/${id}`);
+  const isLoading = !data && !error;
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { handleSubmit, register, reset } = useForm<FormData>({
+    defaultValues: data,
+  });
+
+  useEffect(() => {
+    if (!data) {
+      return;
+    }
+    reset(data);
+  }, [data, reset]);
+
+  const submit = useCallback(
+    async (formData) => {
+      if (!data) {
+        return;
+      }
+
+      try {
+        const updatedApiResource = await ky
+          .patch(`/api/resources/${data.id}`, { json: formData })
+          .json<Resource>();
+        void mutate(updatedApiResource);
+      } catch (error: unknown) {
+        console.error(error);
+      }
+    },
+    [data, mutate]
+  );
+
+  return (
+    <div className={styles.container}>
+      <BackLink to="/api-resources">{t('api_resource_details.back_to_api_resources')}</BackLink>
+
+      {isLoading && <div>loading</div>}
+      {error && <div>{`error occurred: ${error.metadata.code}`}</div>}
+      {data && (
+        <>
+          <Card className={styles.header}>
+            <div className={styles.info}>
+              <ImagePlaceholder size={76} borderRadius={16} />
+              <div className={styles.meta}>
+                <div className={styles.name}>{data.name}</div>
+                <CopyToClipboard value={data.indicator} />
+              </div>
+            </div>
+            <div className={styles.operation}>
+              <Button title="admin_console.api_resource_details.check_help_guide" />
+            </div>
+          </Card>
+          <Card className={styles.body}>
+            <TabNav>
+              <TabNavLink href={location.pathname}>{t('api_resource_details.settings')}</TabNavLink>
+            </TabNav>
+            <form className={styles.form} onSubmit={handleSubmit(submit)}>
+              <div className={styles.fields}>
+                <FormField
+                  isRequired
+                  title="admin_console.api_resources.api_name"
+                  className={styles.textField}
+                >
+                  <TextInput {...register('name', { required: true })} />
+                </FormField>
+                <FormField
+                  isRequired
+                  title="admin_console.api_resource_details.token_expiration_time_in_seconds"
+                  className={styles.textField}
+                >
+                  <TextInput {...register('accessTokenTtl', { required: true })} />
+                </FormField>
+              </div>
+              <div className={styles.submit}>
+                <Button
```

Comment:
do we need a submitting state to prevent unexpected click?

Reply:
can simply set to `disabled` while submitting

Reply:
Good idea.
updated.